_posts/2020-04-07-how-to-start-with-opentelemetry-in-nodejs.markdown

```diff
@@ -0,0 +1,300 @@
+
+```
+$ tree -L 1 ./examples/
+./examples/
+├── basic-tracer-node
+├── dns
+├── express
+├── grpc
+├── grpc_dynamic_codegen
+├── http
+├── https
+├── ioredis
+├── metrics
+├── mysql
+├── opentracing-shim
+├── postgres
+├── prometheus
+├── redis
+└── tracer-web
+
+$ tree -L 1 ./packages/
+./packages/
+├── opentelemetry-api
+├── opentelemetry-base
+├── opentelemetry-context-async-hooks
+├── opentelemetry-context-base
+├── opentelemetry-context-zone
+├── opentelemetry-context-zone-peer-dep
+├── opentelemetry-core
+├── opentelemetry-exporter-collector
+├── opentelemetry-exporter-jaeger
+├── opentelemetry-exporter-prometheus
+├── opentelemetry-exporter-zipkin
+├── opentelemetry-metrics
+├── opentelemetry-node
+├── opentelemetry-plugin-dns
+├── opentelemetry-plugin-document-load
+├── opentelemetry-plugin-express
+├── opentelemetry-plugin-grpc
+├── opentelemetry-plugin-http
+├── opentelemetry-plugin-https
+├── opentelemetry-plugin-ioredis
+├── opentelemetry-plugin-mongodb
+├── opentelemetry-plugin-mysql
+├── opentelemetry-plugin-postgres
+├── opentelemetry-plugin-redis
+├── opentelemetry-plugin-user-interaction
+├── opentelemetry-plugin-xml-http-request
+├── opentelemetry-propagator-jaeger
+├── opentelemetry-resources
+├── opentelemetry-shim-opentracing
+├── opentelemetry-test-utils
+├── opentelemetry-tracing
+├── opentelemetry-web
+└── tsconfig.base.json
+```
+
+The suffix of the package helps you to figure out what they are:
+
+* `opentelemetry-plugin-*` usually contains the code that instrument a specific
+  library, you can see here `express`, `http`, `https`, `dns`. Some plugins are
+  loaded by the `NodeTracerProvider` by default. Other has to be specified. You
+  can to relay on the code or read the documentation to figure it out. For
+  example `http` is loaded by default but if you need `express` you have to load
+  them up by yourself, figuring out the right dependencies. At least or now.
+* `opentelemetry-exporter-*` contains various exporters for now Jaeger,
+  Prometheus, Zipkin the and otel-collector.
+
+Anyway, what I am trying to say is that it is very intuitive and looking here it
+is clear what you can get from this project.
+
+## Plugin
+
+NodeJS sounds very easy to instrument and on the right path to get automatic
+instrumentation right,  because you can listen from the outside to function
+call, you do not need to specifically change your code where you do a request or
+where you get one, you can add tracing in a centralized location. That's how the
+provided plugins work.
+
+[Shimmer](https://github.com/othiym23/shimmer) is the library that simplify the
+trick. I recently had a chat with [Walter](https://twitter.com/walterdalmut)
+because I know he works in NodeJS and during the experiments otel were easy
+enough to fit his use case. He is currently trying it and as he discovered that
+[mongoose](https://github.com/Automattic/mongoose), the ORM library he uses does
```

Comment:
It was in the initial release of the plugin before but we shipped a fix in https://github.com/open-telemetry/opentelemetry-js/pull/652 that should fixes the issue